libcaf_core/caf/cow_tuple.test.cpp

```diff
@@ -0,0 +1,139 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/cow_tuple.hpp"
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/scenario.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/binary_deserializer.hpp"
+#include "caf/binary_serializer.hpp"
+
+using std::make_tuple;
+using std::string;
+using std::tuple;
+
+using namespace caf;
+
+namespace {
+
+struct fixture : test::fixture::deterministic {
+  template <class... Ts>
+  caf::byte_buffer serialize(const Ts&... xs) {
+    caf::byte_buffer buf;
+    caf::binary_serializer sink{sys, buf};
```

Comment:
We don't really need the fixture here. You can turn these three functions into free functions and pass `nullptr` instead of `sys`.